subsys/net/ip/net_if.c

```diff
@@ -4732,6 +4733,147 @@ struct net_if *net_if_get_first_wifi(void)
 	return NULL;
 }
 
+int net_if_get_name(struct net_if *iface, char *buf, int len)
+{
+#if defined(CONFIG_NET_INTERFACE_NAME)
+	int name_len;
+
+	if (iface == NULL || buf == NULL || len <= 0) {
+		return -EINVAL;
+	}
+
+	name_len = strlen(net_if_get_config(iface)->name);
+	if (name_len > len) {
+		return -ERANGE;
+	}
+
+	strncpy(buf, net_if_get_config(iface)->name, name_len);
+
+	return name_len;
+#else
+	return -ENOTSUP;
+#endif
+}
+
+int net_if_set_name(struct net_if *iface, char *buf)
+{
+#if defined(CONFIG_NET_INTERFACE_NAME)
+	int name_len;
+
+	if (iface == NULL || buf == NULL) {
+		return -EINVAL;
+	}
+
+	name_len = strlen(buf);
+	if (name_len > CONFIG_NET_INTERFACE_NAME_LEN) {
+		return -ENAMETOOLONG;
+	}
+
+	memset(net_if_get_config(iface)->name, 0,
+	       CONFIG_NET_INTERFACE_NAME_LEN + 1);
```

Comment:
Maybe use `sizeof(net_if_get_config(iface)->name)` instead of `CONFIG_NET_INTERFACE_NAME_LEN + 1` so there's no need to carry +1 here and there?

Reply:
Unfortunately `sizeof()` is done at compile time so the parameter cannot be a function call.

Reply:
But I think we could access the config directly instead of using a function, I will try that.

Reply:
True, I overlooked this involves a function call within `sizeof()`. If this doesn't fly, then no big deal, it isn't a major issue.